roslibrust_codegen/src/utils.rs

```diff
@@ -145,6 +152,44 @@ fn message_files_from_path(path: &Path, ext: &str) -> io::Result<Vec<PathBuf>> {
     Ok(msg_files)
 }
 
+pub fn deduplicate_packages(packages: Vec<Package>) -> Vec<Package> {
```

Comment:
Hrrmmm I'm slightly wondering if de-duplication at the package level is sane...

Like is it a valid use case to have

my_pkg/MessageA.msg and my_pkg/MessageB.msg and to actually want both to generate?

What you did is almost certainly fine, but has me wondering...

Reply:
Where my_pkg has a different directory in each case with a package of the same name?